linkerd/app/outbound/src/sidecar.rs

```diff
@@ -34,19 +35,34 @@ struct HttpSidecar {
 // === impl Outbound ===
 
 impl Outbound<()> {
-    pub fn mk_sidecar<T, I, P, R>(&self, profiles: P, resolve: R) -> svc::ArcNewTcp<T, I>
+    pub fn mk_sidecar<T, I, R>(
+        &self,
+        profiles: impl profiles::GetProfile<Error = Error>,
+        policies: impl policy::GetPolicy,
+        resolve: R,
+    ) -> svc::ArcNewTcp<T, I>
     where
         // Target describing an outbound connection.
         T: svc::Param<OrigDstAddr>,
         T: Clone + Send + Sync + 'static,
         // Server-side socket.
         I: io::AsyncRead + io::AsyncWrite + io::Peek + io::PeerAddr,
         I: Debug + Unpin + Send + Sync + 'static,
-        // Route discovery.
-        P: profiles::GetProfile<Error = Error>,
         // Endpoint resolver.
         R: Resolve<ConcreteAddr, Endpoint = Metadata, Error = Error>,
     {
+        let discover = {
+            let detect_timeout = self.config.proxy.detect_protocol_timeout;
+
+            // TODO(eliza): opaque should have a different queue config...
+            let queue = self.config.http_request_queue;
+            let default_queue = policy::Queue {
+                capacity: queue.capacity,
+                failfast_timeout: queue.failfast_timeout,
+            };
+            discover::resolver(profiles, policies, default_queue, detect_timeout)
```

Comment:
Mmmm... Is this an HTTP request queue in any of the cases? Isn't this always a tcp connection queue?

Reply:
whoops i think you're right. fixed.